### DISCUSSION TITLE
Patch points

### DISCUSSION THREAD

**Discussion by newtraldev** [qdrant/qdrant#5177] [Ideas]
Hello,
Currently it is only possible to update points using Upsert points this method is a PUT, a new functionality would be to implement a new method using PATCH this would allow to update specific properties without having to send the whole object with all the properties.

What do you think about the possibility of implementing a new method using PATCH, is it possible?

**Comment by timvisee** [comment]
This is possible, but through a different interface.

To update vector(s) of a point you can use: https://qdrant.tech/documentation/concepts/points/#update-vectors

To update the payload of a point you can use any of the described methods here: https://qdrant.tech/documentation/concepts/payload/#update-payload

**Reply by newtraldev** [reply]
Hello,

But if you use Update Payload, you can see https://api.qdrant.tech/api-reference/points/overwrite-payload that this method replaces the whole payload of a specified point with a new payload, so if you only have to change one property, it is mandatory to send the whole payload, it is not the same, with PATCH it is possible to only change one property without sending the whole payload object.

**Reply by timvisee** [reply]
That's why we also have `set_payload` which you can use to update only part of the payload structure.

Imagine having this:

```json
{
    "property1": {
        "nested_property": "foo",
    },
    "property2": {
        "nested_property": "bar",
    }
}
```

You could just update property1 like:

```json
POST /collections/{collection_name}/points/payload
{
    "payload": {
        "nested_property": "qux",
    },
    "key": "property1",
    "points": [1]
}
```

To get this:

```json
{
    "property1": {
        "nested_property": "qux",
    },
    "property2": {
        "nested_property": "bar",
    }
}
```

Documentation: https://qdrant.tech/documentation/concepts/payload/#set-payload
API reference: https://api.qdrant.tech/api-reference/points/set-payload

**Reply by newtraldev** [reply]
Hello, 

Thank you very much, it's my mistake. I didn't see this sentence "It is possible to modify only a specific key of the payload by using the key parameter.", in the documentation of Set Payload. 
I was looking for a PATCH method, as usual in REST Api.
